target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata

```diff
@@ -92,7 +92,8 @@ case "$FIRMWARE" in
 		caldata_extract "ART" 0x1000 0x2f20
 		ath10k_patch_mac $(macaddr_add $(mtd_get_mac_ascii u-boot-env ethaddr) 2)
 		;;
-	extreme-networks,ws-ap3915i)
+	extreme-networks,ws-ap3915i|\
```

Comment:
This should be replaced with NVMEM since now ASCII MAC-s are supported as well.

Same for others as well.

Reply:
I was just copying from the previous 3915 example.. will look into the new way...

Reply:
Hi @robimarko! I'm having trouble finding any examples from recent commits that show setting radio MACADDR from arbitrarily named u-boot variables.  My AP's use RADIOADRR0 and RADIOADDR1.  Can you point me to an example how this is done?

Reply:
Oh yeah, it uses custom names, I am not sure whether that will work as the env parser driver expects ethaddr name for it to post-process it.

Try something like this:
https://github.com/openwrt/openwrt/blob/main/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts#L222C16-L222C16

But in your case, remove the cells property and change the variable name from `ethaddr` to the `RADIOADRR0`, etc.

Also, make sure that the U-Boot parsers is enabled on the target for starters.

Reply:
I was only able to find kernel code that kind of parses it in kernel 6.1 u-boot-env.c u_boot_env_add_cells() function.  And like you said, it only works for env variable named "ethaddr".  No support for any other named variable and I can't find it in kernel 5.15 which this platform still uses.

Can we let this issue go until kernel is ready?

Reply:
OpenWrt has NVMEM stuff backported up to and including 6.6 for both 6.1 and 5.15 so that is not the issue.

This is fine, I am checking with Rafal whether we can extend the driver to work for non standard variables as well.